apps/web/src/utils/logger.ts

```diff
@@ -20,16 +26,41 @@ class CustomLogger {
     return CustomLogger.instance;
   }
 
-  private log(level: LogLevel, message: string, meta?: Record<string, unknown>) {
+  private createDatadogLog(level: LogLevel, message: string, meta?: Record<string, unknown>) {
+    let traceId: string | undefined;
+    let spanId: string | undefined;
+
+    //TODO: initialice ddTrace through dd-tracer
+    if (ddTrace) {
+      // Access trace information server-side
+      const currentSpan = ddTrace?.scope().active();
+      traceId = currentSpan?.context().toTraceId() ?? undefined;
+      spanId = currentSpan?.context().toSpanId() ?? undefined;
+    }
+
+    const logEntry = {
+      message: `[${this.service}] ${message}`,
+      level,
+      dd: {
+        trace_id: traceId,
+        span_id: spanId,
+      },
+      ...meta,
+    };
+
     if (level === 'debug' || level === 'verbose' || level === 'info') {
-      console.log(`[${this.service}] ${message}`, meta);
+      console.log(JSON.stringify(logEntry));
     } else if (typeof console[level] === 'function') {
-      console[level](`[${this.service}] ${message}`, meta);
+      console[level](JSON.stringify(logEntry));
     } else {
-      console.log(`[${this.service}] ${message}`, meta);
+      console.log(JSON.stringify(logEntry));
```

Comment:
do we not want to use `console.error()` or `console.warn()` when those are the levels?

Reply:
otherwise can we structure this as
```
if (typeof console[level] === 'function') {
      console[level](JSON.stringify(logEntry));
else {
  console.log(JSON.stringify(logEntry));
}
```
?

Reply:
I'm using the previous condition for that. there is a warn and error function so it will be going to that. This else in particular is kind of overkill, everything should fall between the first other 2 conditions.